packages/remix-cloudflare-pages/worker.ts

```diff
@@ -0,0 +1,96 @@
+import type { ServerBuild, AppLoadContext } from "@remix-run/server-runtime";
+import { createRequestHandler as createRemixRequestHandler } from "@remix-run/server-runtime";
+
+export interface CreateRequestHandlerParams<Env = any> {
+  build: ServerBuild;
+  getLoadContext?: (context: EventContext<Env, any, any>) => AppLoadContext;
+  mode?: string;
+}
+
+function createInternalHandler<Env = any>({
+  build,
+  getLoadContext,
+  mode
+}: CreateRequestHandlerParams<Env>): PagesFunction<Env> {
+  let platform = {};
+  let handleRequest = createRemixRequestHandler(build, platform, mode);
+
+  return context => {
+    let loadContext =
+      typeof getLoadContext === "function"
+        ? getLoadContext(context)
+        : undefined;
+    return handleRequest(context.request, loadContext);
+  };
+}
+
+type EnvWithAssets = Record<string, unknown> & {
+  ASSETS: { fetch: typeof fetch };
+};
+
+async function handleAsset(request: Request, env: unknown) {
+  let envWithAssets = env as EnvWithAssets;
+  if (!envWithAssets?.ASSETS?.fetch) {
```

Comment:
This check isn't required. We'll always populate `env.ASSETS.fetch`, and you're casting its type above anyway.